README.md

```diff
@@ -217,134 +284,253 @@ spec:
+    tf_replica_type: MASTER
+  - ReplicasStates:
+      Running: 1
+    state: Running
+    tf_replica_type: WORKER
+  - ReplicasStates:
+      Running: 2
+    state: Running
+    tf_replica_type: PS
+  state: Succeeded
 
-The semantics for aggregating status of individual replicas into overall TfJob status needs to be thought out.
+```
 
-### Dead/Unnecessary code
+The first thing to note is the **RuntimeId**. This is a random unique
+string which is used to give names to all the K8s resouces
+(e.g Job controllers & services) that are created by the TfJob.
 
-There is a lot of code from earlier versions (including the ETCD operator) that still needs to be cleaned up.
+As with other K8s resources status provides information about the state
+of the resource.
 
-### Testing
+**phase** - Indicates the phase of a job and will be one of
+ - Creating
+ - Running
+ - CleanUp
+ - Failed
+ - Done
 
-There is minimal testing.
+**state** - Provides the overall status of the job and will be one of
+  - Running
+  - Succeeded
+  - Failed
 
-#### Unittests
+For each replica type in the job, there will be a ReplicaStatus that
+provides the number of replicas of that type in each state.
 
-There are some unittests.
+For each replica type, the job creates a set of K8s
+[Job Controllers](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/)
+named
 
-#### E2E tests
+```
+${REPLICA-TYPE}-${RUNTIME_ID}-${INDEX}
+```
 
-The helm package provides some basic E2E tests.
+For example, if you have 2 parameter servers and runtime id 76n0 TfJob
+will create the jobs
 
-## Building the Operator
+```
+ps-76no-0
+ps-76no-1
+```
 
-Create a symbolic link inside your GOPATH to the location you checked out the code
+### TensorFlow Logs
 
-    ```
-    mkdir -p ${GOPATH}/src/github.com/jlewi
-    ln -sf ${GIT_TRAINING} ${GOPATH}/src/mlkube.io
-    ```
+Logging follows standard K8s logging practices.
 
-  * GIT_TRAINING should be the location where you checked out https://github.com/jlewi/mlkube.io
+You can use kubectl to get standard output/error for any of
+your containers.
 
-Resolve dependencies (if you don't have glide install, check how to do it [here](https://github.com/Masterminds/glide/blob/master/README.md#install))
+First find the pod created by the job controller for the replica of
+index. Pods will be named
 
 ```
-glide install
+${REPLICA-TYPE}-${RUNTIME_ID}-${INDEX}-${RADNOM}
 ```
 
-Build it
+where RANDOM is a unique id generated by K8s to uniquely identify each
+pod.
+
+Once you've identified your pod you can get the logs using kubectl.
 
 ```
-go install github.com/jlewi/mlkube.io/cmd/tf_operator
+kubectl logs ${REPLICA-TYPE}-${RUNTIME_ID}-${INDEX}-${RADNOM}
 ```
 
-## Runing the Operator Locally
+If your cluster takes advantage of K8s
+[logging infrastructure](https://kubernetes.io/docs/concepts/cluster-administration/logging/)
+then your logs may also be shipped to an appropriate data store for
+further analysis.
+
+#### GKE
 
```

Comment:
Because this is not something specific to TfJob, is it worth mentionning?
I'm guessing this is already documented on K8s or Stackdriver doc?

Reply:
I think users will want to know how to get the relevant logs for a TFJob; i.e. how to get the logs for the master? worker i? etc...

So I think the docs should explain that for relevant clouds (feel free to put together a PR with the same for Azure). 

Unfortunately, the story on GCP leaves something to be desired because you can't filter by pod labels currently. Eventually once that's fixed  and #72 is fixed users will be to just filter by various pod labels and I will update the docs to explain relevant filter expressions for TfJob.

I think that's relevant to using TfJob effectively.